<launch>

  # 옵션
  <arg name="camera_hz" default="5"/>
  <arg name="cam" default="true"/>

  # 이동로봇 상태 관리
  <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find shared_control)/urdf/turtlebot3_waffle_gamma.urdf.xacro'"/>
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="state_publisher"/>
  <node pkg="turtlebot3_bringup" type="turtlebot3_diagnostics" name="turtlebot3_diagnostics"/>

  # OpenCR 연결
  <node pkg="rosserial_python" type="serial_node.py" name="turtlebot3_core">
    <param name="port" value="/dev/ttyACM0"/>
    <param name="baud" value="115200"/>
    <remap from="imu" to="imu_raw"/>
  </node>
  <node pkg="shared_control" type="filter.py" name="filter">
    <param name="imu_in" value="imu_raw"/>
    <param name="imu_out" value="imu"/>
    <param name="imu_delayed" value="true"/>
  </node>

  # Realsense 연결
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" if="$(arg cam)">
    <param name="pixel_format" value="yuyv"/>
    <param name="camera_frame_id" value="camera_link"/>
  </node>
  <node pkg="topic_tools" type="throttle" name="throttler" args="messages /usb_cam/image_raw $(arg camera_hz) /camera/color/image"  if="$(arg cam)"/>

</launch>
